### DISCUSSION TITLE
Closing Terminal tab doesn't terminate process

### DISCUSSION THREAD

**Discussion by profbbrown** [eclipse-theia/theia#13341] [Q&A]
Let's say I run a program that produces infinite output. For example, the `yes` command, or a program that goes into an infinite loop.

If I close the Terminal, the process continues running the background. In other words, closing the Terminal does not appear to send the SIGHUP or SIGTERM signal to the process. Is this something I can change in a configuration, or is it an issue that needs fixing?

As far an I can remember, older versions of Theia's terminal (from a couple years ago) terminated the process when the tab was closed.

**Comment by JonasHelming** [comment]
If this is the case, I would say it is a bug, WDYT @tsmaeder ?

**Comment by tsmaeder** [comment]
I would say that depends on the details of how the program was started, but probably yes. Best way of action would be to open an issue with detailed "steps to reproduce". @profbbrown

**Reply by sdirix** [reply]
I tried it in the browser example application of Theia:
* Open a terminal
* Execute `yes`
* Close the terminal
* The terminal can't be reopened but on the host system the `yes` is still running, consuming resources

**Comment by profbbrown** [comment]
Opened as #13346
